test/testutils/syncenginetestutils.cpp

```diff
@@ -41,65 +41,123 @@ PathComponents PathComponents::subComponents() const &
-    QVERIFY(file.exists());
-    file.open(QFile::WriteOnly);
-    file.write(QByteArray {}.fill(contentChar, newSize));
+    _processArguments.append({ "contents", relativePath, QString::number(newSize), QChar::fromLatin1(contentChar) });
 }
 
 void DiskFileModifier::appendByte(const QString &relativePath, char contentChar)
 {
-    QFile file { _rootDir.filePath(relativePath) };
-    QVERIFY(file.exists());
-    file.open(QFile::ReadWrite);
-    QByteArray contents;
-    if (contentChar)
-        contents += contentChar;
-    else
-        contents = file.read(1);
-    file.seek(file.size());
-    file.write(contents);
+    _processArguments.append({ "appendbyte", relativePath, QChar::fromLatin1(contentChar) });
 }
 
 void DiskFileModifier::mkdir(const QString &relativePath)
 {
-    _rootDir.mkpath(relativePath);
+    _processArguments.append({ "mkdir", relativePath });
 }
 
 void DiskFileModifier::rename(const QString &from, const QString &to)
 {
-    QVERIFY(_rootDir.exists(from));
-    QVERIFY(_rootDir.rename(from, to));
+    _processArguments.append({ "rename", from, to });
 }
 
 void DiskFileModifier::setModTime(const QString &relativePath, const QDateTime &modTime)
 {
-    OCC::FileSystem::setModTime(_rootDir.filePath(relativePath), OCC::Utility::qDateTimeToTime_t(modTime));
+    _processArguments.append({ "mtime", relativePath, QString::number(modTime.toSecsSinceEpoch()) });
+}
+
+class HelperProcess : public QProcess
+{
+public:
+    HelperProcess(const QDir &rootDir, QStringList processArguments) // copy is on purpose
+    {
+        if (processArguments.isEmpty()) { // fast-path:
+            finished = true;
+            succeeded = true;
+            return;
+        }
+
+        setProcessChannelMode(QProcess::MergedChannels);
+        processArguments.prepend(rootDir.absolutePath());
+        QObject::connect(this, &QProcess::readyRead, [this]() {
+            while (canReadLine()) {
+                qDebug() << "helper output:" << readLine();
+            }
+        });
+        QObject::connect(this, QOverload<int, QProcess::ExitStatus>::of(&QProcess::finished),
+            [this](int exitCode, QProcess::ExitStatus exitStatus) {
+                qDebug() << "helper finished:" << exitCode << exitStatus;
+                this->finished = true;
+                this->succeeded = exitStatus == QProcess::NormalExit && exitCode == 0;
+            });
+
+        qDebug() << "Starting helper:" << TEST_HELPER_EXE << processArguments;
+        start(TEST_HELPER_EXE, processArguments);
+    }
+
+public:
+    bool finished = false;
+    bool succeeded = true;
+};
+
+bool DiskFileModifier::applyModifications()
+{
+    if (_processArguments.isEmpty()) {
+        return true;
+    }
+
+    HelperProcess helper(_rootDir, _processArguments);
+    helper.waitForStarted();
+    _processArguments.clear(); // the helper has a copy, we clean it now for the next run
+    do {
+        QThread::msleep(50);
```

Comment:
please use 50ms, as QThread has no overload for that  std::this_thread::sleep_for(50ms)
```suggestion
        std::this_thread::sleep_for(50ms);
```